mailgun/route_handlers.py

```diff
@@ -180,32 +180,66 @@ def _handle_recipient(request, recipient):
     # out the address from the display name.
     parsed_sender = address.parse(sender)
     sender_address = parsed_sender.address.lower()
+    from_address = parsed_from.address.lower() if parsed_from else None
+
+    # email to use as reply-to / sender; defaults to sender address, but if the
+    # from address is the actual list member and the sender address is an active
+    # communication channel in Canvas for said member we will use the from
+    # address instead (see logic below)
+    parsed_reply_to = parsed_sender
 
     # any validation that fails will set the bounce template
     bounce_back_email_template = None
 
-    # if not a super sender, check list membership and permissions
-    if sender_address not in super_senders:
-        if ml.access_level == MailingList.ACCESS_LEVEL_EVERYONE:
-            pass
-        elif sender_address not in staff_plus_members:
-            # NOTE: list access is minimally ACCESS_LEVEL_MEMBERS at this point,
-            #       so we want to let them know the email they sent from isn't
-            #       a member of the list.
-            logger.info(
-                u'Sending mailing list bounce back email to %s for mailing list %s '
-                u'because the sender was not a member', sender, recipient)
-            bounce_back_email_template = 'mailgun/email/bounce_back_not_subscribed.html'
-        elif ml.access_level == MailingList.ACCESS_LEVEL_READONLY:
-            logger.info(
-                u'Sending mailing list bounce back email to %s for mailing list %s '
-                u'because the list is readonly', sender, recipient)
-            bounce_back_email_template = 'mailgun/email/bounce_back_readonly_list.html'
-        elif ml.access_level == MailingList.ACCESS_LEVEL_STAFF and sender_address not in teaching_staff_addresses:
-            logger.info(
-                u'Sending mailing list bounce back email to %s for mailing list %s '
-                u'because the sender was not a staff member', sender, recipient)
-            bounce_back_email_template = 'mailgun/email/bounce_back_access_denied.html'
+    # if sender is a super sender or the list permission is open to everyone,
+    # proceed without checking for list membership
+    if sender_address not in super_senders \
+            and ml.access_level != MailingList.ACCESS_LEVEL_EVERYONE:
```

Comment:
Should we also be sending as 'from' in the case of `ml.access_level == MailingList.ACCESS_LEVEL_EVERYONE` (regardless of whether the sender or from address are list members)?